### PR TITLE
add README for cprover binary

### PR DESCRIPTION
**Pull request by kroening**
This adds a README for the cprover binary, which includes instructions for running two sets of verification problems.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/cprover/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+\ingroup module_hidden
+\defgroup cprover cprover
+
+# Folder CPROVER
+
+This contains an experimental encoding from C programs into a Contstrained
```
Comment (peterschrammel) [inline src/cprover/README.md]:
```suggestion
This contains an experimental encoding from C programs into a Constrained
```

File: src/cprover/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Horn Clause (CHC) system.  The satisfiability of the CHC system implies that
+the assertions (as given or generated) hold.  The executable performs the
+encoding and the solving of the CHC system.
+
+The forumla generated uses custom operators that model program heaps,
```
Comment (peterschrammel) [inline src/cprover/README.md]:
```suggestion
The formula generated uses custom operators that model program heaps,
```

**Review by peterschrammel — APPROVED** [review]

File: src/cprover/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# coreJSON
+
+```sh
+git clone https://github.com/kroening/coreJSON
+cd coreJSON
+git checkout cprover
+cd test/cprover
+./setup
+./script
+```
```
Comment (tautschnig) [inline src/cprover/README.md:39]: The layers of scripts in those projects almost obfuscate that what a user has to do is actually really simple?! Perhaps a tiny example could be included in the README file? One problem we seem to have is that there are regression tests, but CI is not going to run them by default (for the `cprover` folder is not included in the parent folders Makefile), so I'm not sure these regression tests currently pass? Else they might well do the trick of serving as an example.

Comment (kroening) [reply src/cprover/README.md:39]: Tests enabled via #7180.

**Pull request merged**